packages/ui/primitives/button.tsx

```diff
@@ -53,18 +53,23 @@ export interface ButtonProps
    * Will display the loading spinner and disable the button.
    */
   loading?: boolean;
+
+  /**
+   * The label to show in the button when `isLoading` is true
+   */
+  loadingText?: string;
 }
 
 const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
-  ({ className, variant, size, asChild = false, loading, ...props }, ref) => {
+  ({ className, variant, size, asChild = false, loadingText, loading, ...props }, ref) => {
```

Comment:
I think `loadingText` is the only part of this PR that I'm not too keen on since it can be pretty destructive (read: yeets children).

I'm happy enough handling loading texts within the components with ternaries since it lets us also consider complexities in display there.